### ISSUE TITLE
MD073: Table of Contents Validation does not preserve code ticks

### ISSUE THREAD

**Issue by hofbi** [rvben/rumdl#634]
### rumdl version

v0.1.94

### Installation method

Pre-built binary

### Bug description

MD073 does not preserve code ticks. For a section like

```markdown
### `my header`
```

which is wrapped in code ticks the resulting toc is

```markdown
- my header
```

 without the code ticks

### Steps to reproduce

See above

### Expected behavior

Maintain the code ticks in the TOC

### Actual behavior

Code ticks are removed

### Configuration

```toml
Enabled the rule, else keep the defaults from https://github.com/rvben/rumdl/blob/main/docs/md073.md
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in **v0.1.96**. Generated/validated TOC entries now preserve inline code spans (and emphasis), so a heading like `### ` followed by a code-span title produces a TOC entry that keeps the backticks (e.g. `- [`code`](#code)`) instead of stripping them. Closing as released.
